app/services/collection.py

```diff
@@ -0,0 +1,47 @@
+from typing import Optional, Dict, Any
+
+import requests
+
+
+def find_collection_url(base_url: str, collection_id: str) -> Optional[str]:
+    response = requests.get(base_url)
+    response.raise_for_status()
+    root_data = response.json()
+
+    for link in root_data.get("links", []):
+        href = link.get("href")
+        if (
+            isinstance(href, str)
+            and link.get("rel") == "child"
+            and collection_id in href
+        ):
+            return href
```

Comment:
although this logic is correct, I don't see a reason to no request directly `{base_url}/collections/{collection_id}` if the collection doesn't exist it will return an error